lib/Internal/functions.php

```diff
@@ -135,17 +144,55 @@ function parseUri(string $uri): array {
 }
 
 /** @internal */
-function onCryptoWatchReadability($watcherId, $socket, $cbData) {
-    /** @var \Amp\Deferred $deferred */
-    list($deferred, $method) = $cbData;
-    $result = \stream_socket_enable_crypto($socket, $enable = true, $method);
+function onCryptoWatchReadability($watcherId, $socket, $data) {
+    /** @var Deferred $deferred */
+    /** @var TlsContext $tlsContext */
+    list($deferred, $tlsContext) = $data;
+
+    $cryptoMethod = $tlsContext->toStreamCryptoMethod(TlsContext::CLIENT);
+    $result = \stream_socket_enable_crypto($socket, $enable = true, $cryptoMethod);
+
     if ($result === true) {
         Loop::cancel($watcherId);
         $deferred->resolve($socket);
-    } elseif ($result === false) {
+    } else if ($result === false) {
```

Comment:
Because I removed it and reduced it to `if ($result) { ... } else { ... }` and then noticed that it can also return `0` and manually added it back.